### PR TITLE
Add usagestats ping for migrated extension metrics

### PR DESCRIPTION
**Pull request by philipp-spiess**
Closes #40257

Based on [RFC 738](https://docs.google.com/document/d/1FNs29PmFrQu6Ns7jspNsfMYDWDl5rBogoUZDBHsD_n0/edit#)

This PR adds aggregation of the new tracking for migrated extensions added via #41317 to the usage statistics ping.

## Test plan

- Check the go test

### CODE REVIEW

File: cmd/frontend/internal/app/updatecheck/handler.go [hunk 1]
```diff
@@ -319,6 +320,7 @@ type pingPayload struct {
 	CodeHostVersions              json.RawMessage `json:"code_host_versions"`
 	CodeHostIntegrationUsage      json.RawMessage `json:"code_host_integration_usage"`
 	IDEExtensionsUsage            json.RawMessage `json:"ide_extensions_usage"`
+	MigratedExtensionsUsage       json.RawMessage `json:"migrated_extensions_usage"`
```
Comment (philipp-spiess) [inline cmd/frontend/internal/app/updatecheck/handler.go:323]: I assume it's enough to add this to the usage ping payload like this and downstream will save it for us somewhere? cc @ebrodymoore

File: client/web/src/site-admin/SiteAdminPingsPage.tsx [hunk 2]
```diff
@@ -377,6 +377,25 @@ export const SiteAdminPingsPage: React.FunctionComponent<React.PropsWithChildren
                     </ul>
                     <ul>Aggregate count of daily redirects from extension to Sourcegraph instance</ul>
                 </li>
+                <li>
+                    Migrated extensions data
+                    <ul>
+                        Aggregate data of:
+                        <li>
+                            <ul>Count interactions with the Git blame feature</ul>
```
Comment (ryankscott) [inline client/web/src/site-admin/SiteAdminPingsPage.tsx:385]: We just need to change Git to git.

File: client/web/src/site-admin/SiteAdminPingsPage.tsx [hunk 3]
```diff
@@ -377,6 +377,25 @@ export const SiteAdminPingsPage: React.FunctionComponent<React.PropsWithChildren
...
+                            <ul>Count of unique users who interacted with the Git blame feature</ul>
```
Comment (ryankscott) [inline client/web/src/site-admin/SiteAdminPingsPage.tsx:386]: s/Git/git

File: doc/admin/pings.md [hunk 4]
```diff
@@ -169,8 +169,18 @@ This telemetry can be disabled using the `disableNonCriticalTelemetry` option in
     - Count of total searches performed
   - Aggregate counts of current daily user state:
     - Count of users who installed the extension
-    - Count of users who uninstalled the extension 
+    - Count of users who uninstalled the extension
   - Aggregate count of current daily redirects from extension to Sourcegraph instance
+- Migrated extensions usage data
+  - Aggregate data of:
+    - Count interactions with the Git blame feature
```
Comment (ryankscott) [inline doc/admin/pings.md:176]: s/Git/git

File: doc/admin/pings.md [hunk 5]
```diff
@@ -169,8 +169,18 @@ This telemetry can be disabled using the `disableNonCriticalTelemetry` option in
...
+    - Count of unique users who interacted with the Git blame feature
```
Comment (ryankscott) [inline doc/admin/pings.md:177]: s/Git/git

**Review by ryankscott — APPROVED** [review]
LGTM - just some caps changes.

File: internal/usagestats/migrated_extensions.go [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+	stats.OpenInEditor = openInEditorUsageByIde
+
+	return &stats, nil
+}
+
+var MigratedExtensionsUsageQuery = `
+	WITH event_log_stats AS (
+		SELECT
+			NULLIF(COUNT(*) FILTER (WHERE name = 'GitBlameEnabled'), 0) :: INT AS git_blame_enabled,
+			NULLIF(COUNT(DISTINCT event_logs.user_id) FILTER (WHERE name = 'GitBlameEnabled'), 0) :: INT AS git_blame_enabled_unique_users,
```
Comment (vdavid) [inline internal/usagestats/migrated_extensions.go:69]: Wow, I feel a bit ashamed of it, but I don't remember ever using `FILTER` in SQL (despite writing **tons** of SQL in my life).

**Review by vdavid — APPROVED** [review]
It looks complete and good to me.

File: internal/usagestats/migrated_extensions.go (hunk 6 above)
Comment (philipp-spiess) [reply internal/usagestats/migrated_extensions.go:69]: Same tbh 😐

File: client/web/src/site-admin/SiteAdminPingsPage.tsx (hunk 2 above)
Comment (philipp-spiess) [reply client/web/src/site-admin/SiteAdminPingsPage.tsx:385]: Isn't Git a name? :D Wikipedia also capitalizes Git in sentences: https://en.wikipedia.org/wiki/Git

Comment (ryankscott) [reply client/web/src/site-admin/SiteAdminPingsPage.tsx:385]: Oh wow TIL, please ignore then :)

**Pull request merged**